scripts/test-api-key-encryption.sh

```
@@ -0,0 +1,100 @@
+    echo "ERROR: App not installed"
+    echo "Install the app first and set an API key"
+    exit 1
+fi
+
+echo "--- DataStore File (Plain Text) ---"
+echo "Location: /data/data/$PKG/files/datastore/settings.preferences_pb"
+echo ""
+
+# Try to read DataStore file
+DATASTORE_CONTENT=$($ADB shell run-as $PKG cat files/datastore/settings.preferences_pb 2>/dev/null)
+if [ $? -eq 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.

(SC2181)